### PR TITLE
Fixes #21321 - Check for foreman_proxy_content to allow puppet upgrade

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: hooks/init/31-upgrade-puppet.rb [hunk 1]
```diff
@@ -52,7 +52,7 @@ def fail_and_exit(message)
   fail_and_exit 'Puppet already installed and upgraded. Skipping.' if File.exist?(PUPPET_UPGRADE_COMPLETE)
 
   katello = Kafo::Helpers.module_enabled?(@kafo, 'katello')
-  foreman_proxy_content = @kafo.param('foreman_proxy_plugin_pulp', 'pulpnode_enabled').value
+  foreman_proxy_content = Kafo::Helpers.module_enabled?(@kafo, 'foreman_proxy_content')
```
Comment (ekohl) [inline hooks/init/31-upgrade-puppet.rb]: I'm surprised we have a copy of that. I'd expect https://github.com/theforeman/kafo/blob/db5645b18cbce377ce089e3a898aee868e965070/lib/kafo/hook_context.rb#L70-L72 to work as well. Have you tried that?

Comment (ehelms) [reply hooks/init/31-upgrade-puppet.rb]: I have not, but I'd prefer to fix this BZ using our proven tooling, then tackle an issue to update usage across the board if thats fine?

Comment (ekohl) [reply hooks/init/31-upgrade-puppet.rb]: Fair enough for now.

Comment (ehelms) [reply hooks/init/31-upgrade-puppet.rb]: Filed a tracking issue -- http://projects.theforeman.org/issues/21324

**Comment by mgamboa** [comment]
Hi Team

Apply the patch but the installer can't finish just give me the following error

Success!
katello-service start --only httpd finished successfully!
Puppet 3 to 4 upgrade initialization complete, continuing with installation
[ INFO 2018-01-25 10:19:28 verbose] Executing hooks in group pre_migrations
[ INFO 2018-01-25 10:19:28 verbose] All hooks in group pre_migrations finished
[ INFO 2018-01-25 10:19:28 verbose] Executing hooks in group boot
[ INFO 2018-01-25 10:19:28 verbose] All hooks in group boot finished
[ INFO 2018-01-25 10:19:28 verbose] Executing hooks in group init
[ INFO 2018-01-25 10:19:28 verbose] Upgrading puppet...
[ INFO 2018-01-25 10:19:28 verbose] Upgrade Step: upgrade_puppet_package...
[ INFO 2018-01-25 10:19:28 verbose] Upgrade Step: stop_services...
[ INFO 2018-01-25 10:19:28 verbose] Upgrade Step: copy_data...
[ INFO 2018-01-25 10:19:28 verbose] Upgrade Step: remove_puppet_port_httpd...
[ INFO 2018-01-25 10:19:28 verbose] Upgrade Step: start_httpd...
[ INFO 2018-01-25 10:19:28 verbose] Puppet 3 to 4 upgrade initialization complete, continuing with installation
[ INFO 2018-01-25 10:19:28 verbose] All hooks in group init finished
[ INFO 2018-01-25 10:19:28 verbose] Loading default values from puppet modules...
[ INFO 2018-01-25 10:19:28 verbose] ... finished
[ INFO 2018-01-25 10:19:28 verbose] Executing hooks in group pre_values
[ INFO 2018-01-25 10:19:28 verbose] All hooks in group pre_values finished
[ INFO 2018-01-25 10:19:28 verbose] Running installer with args "--upgrade-puppet", "-v"
[ INFO 2018-01-25 10:19:28 verbose] Executing hooks in group pre_validations
Resetting puppet params...
[ INFO 2018-01-25 10:19:28 verbose] Resetting puppet params...
[ INFO 2018-01-25 10:19:28 verbose] Installer finished in 0.913510699 seconds
/usr/share/katello-installer-base/hooks/boot/01-helpers.rb:59:in `reset_value': undefined method `unset_value' for nil:NilClass (NoMethodError)
from /usr/share/katello-installer-base/hooks/pre_validations/31-upgrade-puppet.rb:19:in `block (4 levels) in load'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/hooking.rb:34:in `instance_eval'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/hooking.rb:34:in `block (4 levels) in load'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/hook_context.rb:15:in `instance_exec'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/hook_context.rb:15:in `execute'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/hooking.rb:51:in `block in execute'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/hooking.rb:49:in `each'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/hooking.rb:49:in `execute'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/kafo_configure.rb:140:in `execute'
from /usr/share/gems/gems/clamp-1.0.0/lib/clamp/command.rb:68:in `run'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/kafo_configure.rb:119:in `run'
from /usr/share/gems/gems/clamp-1.0.0/lib/clamp/command.rb:133:in `run'
from /usr/share/gems/gems/kafo-2.0.0/lib/kafo/kafo_configure.rb:163:in `run'
from /sbin/foreman-installer:8:in `<main>'

**Comment by ekohl** [comment]
@mgamboa I think this isn't applicable to your version of the installer. Please open a new issue if you have one. Commenting on a merged PR is usually forgotten.

**Pull request merged**